src/plugin.h

```diff
@@ -36,6 +36,11 @@ class Plugin {
 #endif
 
  public:
+  enum class Result {
```

Comment:
As `enum class Result` is used by both the plugins' methods but also by the global hook functions, I am not sure if it belongs in class Plugin. Putting something in a class should either be because of the intention to use its namespace to associate or because of encapsulation. Both is not strictly the case here. The former because `Plugin::Result` sounds strange as it suggests that a plugin (not the handlers) has a result and the latter as the enum class must be public anyway.

I therefore suggest to move the enum class to `namespace kaleidoscope`. Although algernon and Obra seem not to have decided yet what to call the plugins' handler methods, calling the enum class `HandlerResult` or `HookResult` or `CallbackResult` seems more expressive.

Reply:
The members of `enum class Result` are still all caps. That's misleading as to most C++ programmers it will suggest that `OK`, `EVENT_CONSUMED` and `ERROR` are pre-processor macros.

Reply:
To me, `Plugin::Result` suggests that its the result of a plugin task. The reason it is under `Plugin` is because we wanted to tie it to plugins some way, without having to figure out what to call the handler methods :P

Regarding caps: I can be convinced to change that, as they are, indeed, not pre-processor macros.

Reply:
> without having to figure out what to call the handler methods :P

That's the first time that I am confronted with using classes to avoid proper naming :wink:
I will try to remember it for the next time that someone asks me about the reasons for using OOP.